drivers/hmis/spec/factories/hmis/ce/referrals.rb

```diff
@@ -2,7 +2,10 @@
 
 FactoryBot.define do
   factory :hmis_ce_referral, class: 'Hmis::Ce::Referral' do
-    association(:opportunity, factory: :hmis_ce_opportunity)
+    transient do
+      project { nil }
+    end
+    opportunity { build(:hmis_ce_opportunity, project: project) }
```

Comment:
Cool, I didn't know you could do it like this! I usually use `after(:create)` to set the transient properties, this is way neater. Will this work for both build and create?

Reply:
Hm, yeah you are right, this doesn't work well with `create`, which I ran into when writing tests for the changes in `current_step`. I'll update this to match our usual pattern.

(You can replace `build` with `create` here on line 8, but then the factory would always save to the db, and couldn't be used to just make an in-memory object. I was also running into errors with transient properties that were required and couldn't correctly default to the factory association.)